src/test/java/io/camunda/testing/util/Utilities.java

```diff
@@ -92,7 +93,7 @@ public static ProcessInstanceEvent startProcessInstance(
             .variables(variables)
             .send()
             .join();
-    Thread.sleep(100);
+    waitForIdleState(engine);
```

Comment:
Idle state is still a bit vague. We consider the engine to be idle when it is not actively processing. One of the challenges we have is figuring out how to notice this, but that's not relevant to the current PR. It could be as simple as not processing for X amount of time for the MVP.

If a process has no "waiting" elements (timers, user tasks, message events), the engine will complete the entire process. Once the process has been completed the engine does not have any processing to do, thus the engine is considered to be idle.

Reply:
It might be an edge case, but I'm trying to imagine processes without waiting elements that don't in 'reasonable' time. Perhaps we don't have to care about it now 👍